### ISSUE TITLE
Type Error during extracting pages in some pdfs

### ISSUE THREAD

**Issue by psrubing** [pdfminer/pdfminer.six#720]
Hello,

I've encountered bug during extrating pages using extract_pages() function from pdfminer.high_level module. This only happens to some pdf-s. 
Image below provides this bug:

![screenshot](https://user-images.githubusercontent.com/37271385/155157292-c745f29b-fa8f-47cb-855b-3820cfeb4892.PNG)

Below pdf implies this bug:
[pdf_bug.pdf](https://github.com/pdfminer/pdfminer.six/files/8117593/pdf_bug.pdf)

Environment: 
Python - 3.7.11
pdfminer.six - 20201018

**Comment by pietermarsman** [comment]
Can replicate:

```shell
$ PYTHONPATH=. python tools/pdf2txt.py ~/Downloads/pdf_bug.pdf 
Traceback (most recent call last):
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdffont.py", line 920, in char_width
    return cast(Dict[int, float], self.widths)[cid] * self.hscale
KeyError: 67

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdffont.py", line 924, in char_width
    return str_widths[self.to_unichr(cid)] * self.hscale
KeyError: 'a'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pieter/projects/pdfminer-upstream/tools/pdf2txt.py", line 313, in <module>
    sys.exit(main())
  File "/home/pieter/projects/pdfminer-upstream/tools/pdf2txt.py", line 307, in main
    outfp = extract_text(**vars(parsed_args))
  File "/home/pieter/projects/pdfminer-upstream/tools/pdf2txt.py", line 62, in extract_text
    pdfminer.high_level.extract_text_to_fp(fp, **locals())
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/high_level.py", line 121, in extract_text_to_fp
    interpreter.process_page(page)
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdfinterp.py", line 991, in process_page
    self.render_contents(page.resources, page.contents, ctm=ctm)
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdfinterp.py", line 1010, in render_contents
    self.execute(list_value(streams))
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdfinterp.py", line 1036, in execute
    func(*args)
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdfinterp.py", line 896, in do_TJ
    self.device.render_string(
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdfdevice.py", line 133, in render_string
    textstate.linematrix = self.render_string_horizontal(
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdfdevice.py", line 173, in render_string_horizontal
    x += self.render_char(
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/converter.py", line 206, in render_char
    textwidth = font.char_width(cid)
  File "/home/pieter/projects/pdfminer-upstream/pdfminer/pdffont.py", line 926, in char_width
    return self.default_width * self.hscale
TypeError: unsupported operand type(s) for *: 'PDFObjRef' and 'float'
```

**Comment by pietermarsman** [comment]
Probably the solution is to call `resolve1` when getting the default width.

**Comment by psrubing** [comment]
Hi, thanks for response, but I don't understand your comment. Function `extract_pages()` doesn't take any parameter related to `resolve1`  as from my knowledge based on documentation:

![image](https://user-images.githubusercontent.com/37271385/155505710-c42bd735-2592-4b0a-ad7b-213b51e66ad3.png)

Or did I missed something?
Best regards

**Comment by pietermarsman** [comment]
I mean, to fix this issue we have to make a change to pdfminer.six, using `resolve1()`. This is a bug in the current code.

**Comment by psrubing** [comment]
Okey, I understand now :) Do you know approximate time of release with this fix?

**Comment by pietermarsman** [comment]
Nobody is working on it as far as I know. 

Do you have time to work on this?

**Comment by psrubing** [comment]
Unfortunately I don't :/ Have to work on different projects, but if something change I will update and could look at this bug.

**Comment by gosiafilipek** [comment]
![image](https://user-images.githubusercontent.com/90677030/176150083-1e177307-1748-4dae-923e-89f1adbd559f.png)
I found another file that generate similar error.
I already found solution so I will create pull request.

Below pdf implies this bug
[pdf_bug2.pdf](https://github.com/pdfminer/pdfminer.six/files/8999877/pdf_bug2.pdf)

**Comment by datatalking** [comment]
@pietermarsman @psrubing This was an issue for me in a past project and they ended up using an OCR solution. I was going to say I could take a look at this to debug it but noticed @gosiafilipek created a PR solution?

If those are done I have time in the next 2 months to contribute, but didn't see a 'good first issue' icon or whatever its called so I looked back and found these I could start with. Does anyone have requests or recommendations on where I should start?

https://github.com/pdfminer/pdfminer.six/issues/470
https://github.com/pdfminer/pdfminer.six/issues/154
https://github.com/pdfminer/pdfminer.six/issues/499
https://github.com/pdfminer/pdfminer.six/issues/497

**Comment by pietermarsman** [comment]
Hi @datatalking,

Thanks for reaching out! And for wanting to help! You can get in touch on gitter.im. In the private or group chat. We can have a sync about what to work on. 

I'll try and see if I can create a good-first-issue label.

**Comment by pietermarsman** [comment]
Fixed by #772
